lte/gateway/python/magma/mobilityd/ip_allocator_dhcp.py

```diff
@@ -435,30 +439,22 @@ def _release_dhcp_ip(self, ip_desc: IPDesc) -> None:
         dhcp_desc = self.get_dhcp_desc_from_store(mac, vlan)
         logging.info("Releasing dhcp desc: %s", dhcp_desc)
         if dhcp_desc:
-            call_args = [
-                DHCP_HELPER_CLI,
-                "--mac", str(mac),
-                "--vlan", str(vlan),
-                "--interface", self._iface,
-                "--json",
-                "release",
-                "--ip", str(ip_desc.ip),
-                "--server-ip", str(dhcp_desc.server_ip),
-            ]
-            ret = subprocess.run(
-                call_args,
-                capture_output=True,
-                check=False,
+            if len(self.dhcp_helper_procs) == self.MAX_DHCP_PROCS:
+                oldest_proc = self.dhcp_helper_procs.pop()
+                if oldest_proc.poll() is None:
+                    oldest_proc.wait(timeout=10)
```

Comment:
Iiuc this can throw a TimeoutExpired exception and by this prevent the current call from opening a subprocess.

Reply:
I added error handling for `TimeoutExpired`. If the process does not finish within 10 seconds, we kill it and move on.  However, there is still a problem: If we kill the release process and the release did not succeed, we will still have deleted the entry in the MobilityStore. I am not sure how we can mitigate this issue.

Reply:
Independent on how the remaining issue plays out - I suggest to refactor the complete while-part into an additional function.

> If we kill the release process and the release did not succeed, we will still have deleted the entry in the MobilityStore. I am not sure how we can mitigate this issue.

Iiuc then this happens before the `_release_dhcp_ip` call, i.e., we always had the potential that the store and the dhpc (is this only redis or also the dhpc server itself?) are out of sync. The only real mitigation I see here is a regular in-sync check with clean up - but imho this is out of scope here. Do we know of implications when store and redis are out of sync?

Reply:
I added your requested changes in the latest commit.